server/schedule/region_splitter.go

```diff
@@ -72,20 +84,25 @@ func (r *RegionSplitter) splitRegionsByKeys(splitKeys [][]byte, newRegions map[u
 	groupKeys, unProcessedKeys := r.groupKeysByRegion(splitKeys)
 	for regionID, keys := range groupKeys {
 		region := r.cluster.GetRegion(regionID)
-		// TODO: assert region is not nil
-		// TODO: assert leader exists
-		// TODO: assert region replicated
-		// TODO: assert region not hot
+		if region == nil {
+			unProcessedKeys = append(unProcessedKeys, keys...)
+			continue
+		}
+		if !r.checkRegionValid(region) {
+			unProcessedKeys = append(unProcessedKeys, keys...)
+			continue
+		}
 		err := r.handler.SplitRegionByKeys(region, keys)
 		if err != nil {
 			unProcessedKeys = append(unProcessedKeys, keys...)
 			continue
 		}
 		// TODO: use goroutine to run watchRegionsByKeyRange asynchronously
```

Comment:
Does this PR need to include it?

Reply:
I will implement this enhancement in the next PR.